app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -226,4 +227,20 @@ class PatientsScreenController @Inject constructor(
         .filter { it.permissionResult == RuntimePermissionResult.GRANTED }
         .map { Ui::openScanSimpleIdCardScreen }
   }
+
+  private fun handleVisibilityOfSyncIndicator(events: Observable<UiEvent>): Observable<UiChange> {
+    val screenCreated = events.ofType<ScreenCreated>()
+
+    val canUserSync = { userSession.canSyncData().distinctUntilChanged() }
+
+    val userApprovedForSync = Observables.combineLatest(screenCreated, canUserSync())
+        .filter { (_, canUserSync) -> canUserSync }
+        .map { { ui: Ui -> ui.showSyncIndicator() } }
+
+    val userWaitingForApproval = Observables.combineLatest(screenCreated, canUserSync())
+        .filter { (_, canUserSync) -> canUserSync.not() }
+        .map { { ui: Ui -> ui.hideSyncIndicator() } }
+
+    return userApprovedForSync.mergeWith(userWaitingForApproval)
```

Comment:
Hrm, what was the decision behind splitting these two streams? 

I can't see any advantages and maybe we can simplify them:

```kotlin
val screenCreated = events.ofType<ScreenCreated>()

val canUserSync = userSession
    .canSyncData()
    .distinctUntilChanged()

return Observables.combineLatest(screenCreated, canUserSync)
    .map { (_, canSync) ->
      when {
        canSync -> { ui: Ui -> ui.showSyncIndicator() }
        else -> { ui: Ui -> ui.hideSyncIndicator() }
      }
    }
```

Reply:
Just to avoid using imperative code in RxStreams.
Maybe it is not that relevant here.